fake-koji/src/main/java/org/fakekoji/http/PreviewFakeKoji.java

```diff
@@ -789,40 +789,73 @@ public void runImpl() throws IOException {
+                try (OutputStream os = t.getResponseBody()) {
+                    os.write(result.getResponse() .getBytes());
+                }
+            }
+
+            private Result getResult(String parameter) {
+                try {
+                    return new Result(settings.get(parameter), 200);
+                } catch (ProjectMappingExceptions.ProjectMappingException e) {
+                    return new Result(e.getMessage(), 400);
+                }
+            }
+
+            private Result getResultFor(String rawQuery) {
+                final StringBuilder response = new StringBuilder();
+                int resultCode = 200;
                 if (rawQuery == null) {
-                    s = GET_HELP;
+                    response.append(GET_HELP);
                 } else {
                     String[] query = rawQuery.split("&");
                     if (query.length == 0 || rawQuery.trim().isEmpty()) {
-                        s = GET_HELP;
+                        response.append(GET_HELP);
                     } else {
-                        s = getAnswersFor(query);
-                        if (s == null) {
-                            s = "Unknow question in `" + rawQuery + "`. " + GET_HELP;
+                        if (query.length == 1) {
+                            Result result = getResult(query[0]);
+                            response.append(result.getResponse());
+                            resultCode = result.getReturnCode();
                         } else {
-                            returnCode = 200;
+                            for (String key : query) {
+                                Result result = getResult(key);
+                                resultCode = result.getReturnCode() != 200 ? 400 : resultCode;
+                                response.append(key).append("=");
+                                response.append(result.getResponse());
+                                response.append("\n");
+                            }
                         }
-
                     }
                 }
-                long size = s.length(); //yahnot perfect, ets assuemno one will use this on chinese chars
-                t.sendResponseHeaders(returnCode, size);
-                try (OutputStream os = t.getResponseBody()) {
-                    os.write(s.getBytes());
-                }
+                return new Result(response.toString(), resultCode);
             }
 
-            private String getAnswersFor(String[] query) {
-                if (query.length == 1) {
-                    return settings.get(query[0]);
-                } else {
-                    StringBuilder sb = new StringBuilder();
-                    for (String key : query) {
-                        sb.append(key).append("=").append(settings.get(key)).append("\n");
-                    }
-                    return sb.toString();
+            private class Result {
```

Comment:
If possible, pelase rename this to ResponseWithHtmlCode or simialrly. It i (unluckyly) not some plain result, but reastult with strong meaning. I would preffere   bool instead of int, or any other - better enum or similarly and change it to 200/400 later via some method. Leave detaisl of rework here on you. just double check it